sdk/clientcore/http-okhttp3/src/main/java/io/clientcore/http/okhttp3/OkHttpHttpClientBuilder.java

```diff
@@ -354,8 +356,9 @@ public HttpClient build() {
                     buildProxyOptions.getNonProxyHosts()));
 
             if (buildProxyOptions.getUsername() != null) {
-                ProxyAuthenticator proxyAuthenticator = new ProxyAuthenticator(buildProxyOptions.getUsername(),
-                    buildProxyOptions.getPassword());
+                ProxyAuthenticator proxyAuthenticator = new ProxyAuthenticator(ChallengeHandler.of(new DigestProxyChallengeHandler(
```

Comment:
I think we'd be better off replacing username and password in `ProxyOptions` with ChallengeHandler. Or, something a bit more in-depth:

Continue supporting username and password configurations but explicitly state that it will generate a composite ChallengeHandler that supports Basic and Digest authentication. But also add an API which accepts a customer provided ChallengeHandler.

Reply:
Updated the ProxyOptions to internally create a CompositeChallengeHandler with Basic and Digest authentication and also added an API for users to specify a challengehandler.